org.eclipse.jdt.core/compiler/org/eclipse/jdt/internal/compiler/parser/Parser.java

```diff
@@ -10928,9 +10928,10 @@ protected void consumeRecordPatternWithId() {
 	this.astPtr -= length;
 	// Identifier
 	char[] identifierName = this.identifierStack[this.identifierPtr];
-	long namePositions = this.identifierPositionStack[this.identifierPtr--];
-	TypeReference type = getTypeReference(0);
+	long namePositions = this.identifierPositionStack[this.identifierPtr];
 	LocalDeclaration local = createLocalDeclaration(identifierName, (int) (namePositions >>> 32), (int) namePositions);
+    this.identifierPtr--;
```

Comment:
Just wondering if this can be wrapped into the previous statement retrieving the namePositions, like we do in other places, like this:
`long namePositions = this.identifierPositionStack[this.identifierPtr--];`

Reply:
That will not work as indexOfAssistIdentifier will be negative and no valid Java element will be found.